src/plugins/initiatives/initiativesDirectory.js

```diff
@@ -103,3 +106,47 @@ export function _initiativeAddress(
     ...NodeAddress.toParts(_trackerAddress(remoteUrl, fileName))
   );
 }
+
+// Checks the path exists and is a directory.
+// Returns the absolute path or throws.
+export async function _validatePath(localPath: string): Promise<string> {
+  const absPath = path.resolve(localPath);
+  if (!(await fs.exists(absPath))) {
+    throw new Error(
+      `Provided initiatives directory does not exist at: ${absPath}`
+    );
+  }
+  if (!(await fs.lstat(absPath)).isDirectory()) {
+    throw new Error(
+      `Provided initiatives directory is not a directory at: ${absPath}`
+    );
+  }
+  return absPath;
+}
+
+// Gets all *.json filenames in the given directory.
+export async function _findFiles(
+  localPath: string
+): Promise<$ReadOnlyArray<string>> {
+  const absoluteFileNames = await globby(path.join(localPath, "*.json"));
+  return absoluteFileNames.map((a) => path.basename(a));
+}
+
+type NamesToInitiativeFiles = Map<string, InitiativeFile>;
+
+// Reads all given filenames in the given directory, validating them as compat.
+export async function _readFiles(
+  localPath: string,
+  fileNames: $ReadOnlyArray<string>
+): Promise<NamesToInitiativeFiles> {
+  const map: NamesToInitiativeFiles = new Map();
+
+  // TODO(perf): can do this as a stream or in parallel.
+  const sortedFileNames = [...fileNames].sort();
```

Comment:
not that I mind, but why do we sort them?

Reply:
The eventual type we'll arrive at is: `$ReadOnlyArray<Initiative>`.
This means it's an ordered type.

During prototyping I had several test flakes. So I thought, it would be good for reproducibility of loads as well to side with caution and sort them. Rather than sort them in tests only. As this order will be relevant in all parts of the system up to the WeightedGraph. Once in a graph it can be sorted by it's address.

Reply:
> The eventual type we'll arrive at is: $ReadOnlyArray<Initiative>.
This means it's an ordered type.

[] and ROA are both ordered. :)

> During prototyping I had several test flakes. So I thought, it would be good for reproducibility of loads as well to side with caution and sort them. Rather than sort them in tests only. As this order will be relevant in all parts of the system up to the WeightedGraph. Once in a graph it can be sorted by it's address.

Makes sense!

Reply:
Update: I've left a comment about this in the file.

Reply:
Thanks, I appreciate the comment.